src/main/java/org/killbill/billing/plugin/adyen/api/mapping/PaymentInfoMappingService.java

```diff
@@ -134,6 +134,19 @@ private static void set3DSecureFields(final PaymentInfo paymentInfo, final Itera
         final String mpiDataAuthenticationResponse = PluginProperties.findPluginPropertyValue(PROPERTY_MPI_DATA_AUTHENTICATION_RESPONSE, properties);
         paymentInfo.setMpiDataAuthenticationResponse(mpiDataAuthenticationResponse);
 
+        final String mpiDataEci = PluginProperties.findPluginPropertyValue(PROPERTY_MPI_DATA_ECI, properties);
+        paymentInfo.setMpiDataEci(mpiDataEci);
+
+        final String selectedBrand = PluginProperties.findPluginPropertyValue(PROPERTY_SELECTED_BRAND, properties);
+
+        if (selectedBrand != null) {
+            paymentInfo.setMpiDataDirectoryResponse("Y");
+            paymentInfo.setMpiDataAuthenticationResponse("Y");
+            if (mpiDataEci == null || mpiDataEci.isEmpty()) {
+                paymentInfo.setMpiDataEci("7");
```

Comment:
Is it expected the brand overrides this value?

Reply:
Yes, Adyen asked to send these values.